### ISSUE TITLE
Error: cannot open shared object file: No such file or directory

### ISSUE THREAD

**Issue by SilasMarvin** [eisberg-labs/nextjs-node-loader#17]
First, I wanted to say thank you for building this module. We also use Neon and found the same build issues you did. Thank you for building a solution!

Repository of the failed build: https://github.com/SilasMarvin/pgml-nextjs-chatbot
Repository of the node module causing the issue: https://www.npmjs.com/package/pgml

I am having an issue where running and building locally and building on Vercel all give the same error: cannot open shared object file: No such file or directory.

I can get local development and builds working if I remove `__webpack_public_path__` in the index.js file as that seems to add '/_next' to the path. The Vercel builds also work when I remove `__webpack_public_path__` but throw with the same error during runtime.

When I inspect the output of the source on Vercel I am unable to find the .node files where I would normally find them locally. I am not sure what could be causing this, but I am happy to help test anything I can!

**Comment by amarjanica** [comment]
Hi, thanks for creating the reproducible! I'll take a look at it next week and report here what I find

**Comment by amarjanica** [comment]
I had a quick look, your node files seem to be generated here:
```
find . -name x86_64-unknown-linux-gnu-index.node
./.next/server/chunks/x86_64-unknown-linux-gnu-index.node
./node_modules/pgml-test/dist/x86_64-unknown-linux-gnu-index.node
```
Where we're only interested in the first path. Yes, `__webpack_public_path__` is causing the issue.

I haven't been using this package for a while as I moved away the backend part to another framework.
Last time my api was using nextjs, it was  `13.2.4`. Could be that they changed something in 13.4.13.
If I downgrade your example to 13.2.4, it works as expected.

My memory is a little rusty, I'll need a little bit of assistance:
If I setup an option where you can toggle `__webpack_public_path__` would it help?
What I'm not sure and needs to be checked: 
- what is changed since 13.2.4-13.4.13 and is  `__webpack_public_path__` needed at all in newer version?

For example:
```
{
          loader: "nextjs-node-loader",
          options: {
            flags: os.constants.dlopen.RTLD_NOW,
            includeWebpackPublicPath: false, 
            outputPath: config.output.path
          }
        },
```

**Comment by SilasMarvin** [comment]
Thanks for looking into it! That would help for local builds, but even with the `__webpack_public_path__` removed remote builds on Vercel are failing for reasons I am unable to diagnose. I would like to give more assistance on how we can fix the remote builds, but I have not been able to find any information on the differences between remote builds and local builds, and the inspector they give us to view the output of builds on Vercel is less than helpful.

**Comment by amarjanica** [comment]
Hey, sorry for not pinging back. I haven't forgotten about the issue, just that I'm tight with time. 
I've diffed next releases after 13.2.4. and issue starts occurring right on [13.2.5-canary.0](https://www.npmjs.com/package/next/v/13.2.5-canary.0). Removing the _webpack_public_path_ helps, but I wouldn't want to publish before we trace down what happens on your vercel build.
- If you build and deploy with 13.2.4, does this still happen on vercel build?
- Could you add more logs when building to Vercel? I'm interested where does vercel store your node assets.

**Comment by amarjanica** [comment]
I managed to take a look at it. Try the 1.1.5-alpha.0 version. There were a couple of issues:
- load path of the koffi needs to be set in webpack alias, at least in my case. Take a look at [](https://github.com/eisberg-labs/nextjs-node-loader/blob/main/examples/next-13-koffi/next.config.js)
- _webpack_public_path_ causes issues for newer nextjs builds, it's now excluded as default.

I published an alpha version because I haven't properly tested this and have to include more integration tests in github actions.
Let me know if it works.
